File: smoke_tests/swift/command_injection.swift
// Command Injection vulnerabilities in Swift
import Foundation

class CommandInjectionVulnerabilities {
    func executeCommand(userInput: String) -> String {
        // VULNERABLE: Command injection via Process
        let task = Process()
        task.launchPath = "/bin/sh"
        task.arguments = ["-c", "ls \(userInput)"]
        let pipe = Pipe()
        task.standardOutput = pipe
        task.launch()
        return String(data: pipe.fileHandleForReading.readDataToEndOfFile(), encoding: .utf8) ?? ""
    }

    func runShellCommand(command: String) -> String {
        // VULNERABLE: Shell command execution
        let task = Process()
        task.launchPath = "/bin/bash"
        task.arguments = ["-c", command]
        let pipe = Pipe()
        task.standardOutput = pipe
        task.launch()
        return String(data: pipe.fileHandleForReading.readDataToEndOfFile(), encoding: .utf8) ?? ""
    }

    func pingHost(host: String) -> String {
        // VULNERABLE: Command injection in ping
        let task = Process()
        task.launchPath = "/sbin/ping"
        task.arguments = ["-c", "1", host]
        task.launch()
        return ""
    }

    func executeScript(scriptPath: String) {
        // VULNERABLE: Arbitrary script execution
        let task = Process()
        task.launchPath = "/bin/bash"
        task.arguments = [scriptPath]
        task.launch()
    }
}

File: smoke_tests/kotlin/command_injection.kt
// Command Injection vulnerabilities in Kotlin
import java.lang.Runtime

class CommandInjectionVulnerabilities {
    fun executeCommand(userInput: String): String {
        // VULNERABLE: Command injection via Runtime.exec
        val process = Runtime.getRuntime().exec("ls $userInput")
        return process.inputStream.bufferedReader().readText()
    }

    fun runShellCommand(command: String): String {
        // VULNERABLE: Shell command execution
        val process = ProcessBuilder("sh", "-c", command).start()
        return process.inputStream.bufferedReader().readText()
    }

    fun pingHost(host: String): String {
        // VULNERABLE: Command injection in ping
        val cmd = arrayOf("sh", "-c", "ping -c 1 $host")
        val process = Runtime.getRuntime().exec(cmd)
        return process.inputStream.bufferedReader().readText()
    }

    fun executeScript(scriptPath: String) {
        // VULNERABLE: Arbitrary script execution
        Runtime.getRuntime().exec("bash $scriptPath")
    }

    fun grepLogs(pattern: String): String {
        // VULNERABLE: Command injection in grep
        val process = ProcessBuilder("grep", pattern, "/var/log/app.log").start()
        return process.inputStream.bufferedReader().readText()
    }
}
